debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/YugabyteReplicaIdentityIT.java

```diff
@@ -0,0 +1,350 @@
+
+    SourceRecord insertRecord = records.get(0);
+    SourceRecord updateRecord = records.get(1);
+
+    YBVerifyRecord.isValidInsert(insertRecord, PK_FIELD, 1);
+    YBVerifyRecord.isValidUpdate(updateRecord, PK_FIELD, 1);
+
+    Struct updateRecordValue = (Struct) updateRecord.value();
+    assertThat(updateRecordValue.get(Envelope.FieldName.AFTER)).isNotNull();
+    assertThat(updateRecordValue.get(Envelope.FieldName.BEFORE)).isNull();
+
+    // After field will have entries for all the changed columns.
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("pk").getInt32("value")).isEqualTo(1);
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("aa").getInt32("value")).isEqualTo(12345);
+    assertThat(updateRecordValue.getStruct(Envelope.FieldName.AFTER).getStruct("bb")).isNull();
+  }
+
+  @Test
+  public void shouldThrowExceptionWithReplicaIdentityNothingOnUpdatesAndDeletes() throws Exception {
+    /*
+      According to Postgres docs:
+      If a table without a replica identity is added to a publication that replicates
+      UPDATE or DELETE operations then subsequent UPDATE or DELETE operations will cause
+      an error on the publisher.
+
+      Details: https://www.postgresql.org/docs/current/logical-replication-publication.html
+     */
+    TestHelper.execute("ALTER TABLE s2.a REPLICA IDENTITY NOTHING;");
+
+    Configuration config = TestHelper.defaultConfig()
+                             .with(PostgresConnectorConfig.SNAPSHOT_MODE, PostgresConnectorConfig.SnapshotMode.NEVER.getValue())
+                             .with(PostgresConnectorConfig.DROP_SLOT_ON_STOP, Boolean.TRUE)
+                             .build();
+    start(PostgresConnector.class, config);
+    assertConnectorIsRunning();
+
+    // YB Note: Added a wait for replication slot to be active.
+    TestHelper.waitFor(Duration.ofSeconds(10));
+
+    waitForAvailableRecords(10_000, TimeUnit.MILLISECONDS);
+    // there shouldn't be any snapshot records
+    assertNoRecordsToConsume();
+
+    // insert and verify 2 new records
+    TestHelper.execute("INSERT INTO s2.a VALUES (1, 22, 'random text value');");
+
+    try {
+      TestHelper.execute("UPDATE s2.a SET aa = 12345 WHERE pk = 1;");
+    } catch (Exception sqle) {
+      assertThat(sqle.getMessage()).contains("ERROR: cannot update table \"a\" because it does "
+                                              + "not have a replica identity and publishes updates");
+    }
+
+    try {
+      TestHelper.execute("DELETE FROM s2.a WHERE pk = 1;");
+    } catch (Exception sqle) {
+      assertThat(sqle.getMessage()).contains("ERROR: cannot delete from table \"a\" because it "
+                                              + "does not have a replica identity and publishes deletes");
+    }
+  }
+
+  @Test
+  public void shouldHaveBeforeImageForDeletesForReplicaIdentityFull() throws Exception {
```

Comment:
Can we combine this with the test `shouldProduceOldValuesWithReplicaIdentityFull`. This will reduce the code duplication and number of tests in this file.

Reply:
The tests are written in JUnit4 and parameterising the tests there would require an extra effort of adding a separate runner class so I followed the test model Debezium is following in other test files and added duplicate test classes wherever required.